--- a/pkgs/stdenv/linux/default.nix
+++ b/pkgs/stdenv/linux/default.nix
@@ -35,12 +35,14 @@
   # This function downloads a file.
   download = {url, sha1, pkgname}: derivation {
     name = baseNameOf (toString url);
-    builder = bootstrapTools.bash;
-    inherit system curl url;
-    args = [ ./scripts/download.sh ];
+    builder = "builtin:fetchurl";
+    system = "builtin";
+    url = builtins.replaceStrings [ "http://" ] [ "https://" ] url;
     outputHashAlgo = "sha1";
     outputHash = sha1;
+    outputHashMode = "flat";
     impureEnvVars = [ "http_proxy" "https_proxy" "ftp_proxy" "all_proxy" "no_proxy" ];
+    preferLocalBuild = true;
   };
 
   # This function downloads and unpacks a file.
